doc/compiler/il/GlRegDeps.md

```diff
@@ -0,0 +1,152 @@
+n545n             lshl                                                                        [0x7f66870b9a10] bci=[1,29,-] rc=1 vc=0 vn=- li=- udi=- nc=2
+n543n               i2l                                                                       [0x7f66870b9970] bci=[1,29,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n190n                 ==>iand
+n544n               iconst 3                                                                  [0x7f66870b99c0] bci=[-1,0,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n546n             lconst 24                                                                   [0x7f66870b9a60] bci=[-1,0,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n572n     aRegStore ecx                                                                       [0x7f66870ba280] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=1
+n549n       ==>aloadi
+n564n     BBEnd </block_58> ===== ()                                                          [0x7f66870ba000] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=1 flg=0x20
+n562n       GlRegDeps ()                                                                      [0x7f66870b9f60] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=3 flg=0x20
+n506n         ==>aRegLoad
+n505n         ==>iRegLoad
+n587n         PassThrough rcx                                                                 [0x7f66870ba730] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n549n           ==>aloadi
+
+```
+
+The children of the `GlRegDeps` node that represent new values and are assigned to new global registers in the extended block should have been evaluated before `GlRegDeps`. The job of `GlRegDeps` is to take values that are already in virtual registers and simply map them to real registers. `GlRegDeps` is only about generating the register dependency. It should not be about generating any other instructions.
+
+The only other instruction that a `GlRegDeps` evaluation can result in is a register-register move when the same value (same IL node) goes out in two (or more) different global registers. In such a case the same virtual register cannot be mapped to two (or more) real registers in the codegen register dependency since this is a condition that cannot be satisfied. By adding a new virtual register and inserting a register-register copy instruction, the codegen register dependency can use different virtual/real registers. Note that a register-register copy instruction does not result in any arithmetic or setting of control registers. Therefore, it is a safe instruction to execute between the compare and the branch for an `if` opcode where the `GlRegDeps` is present.
+
+The following example of `GlRegDeps` will cause issues. It does the `aload` operations (`n111n` and `n106n`) and the `add` operation (`n110n`) in the middle of laying down register dependencies. These actual tree evaluations interfere with the register dependencies which would end up interfering with branching.
+
```

Comment:
"These actual tree evaluations interfere with the" should probably be "These actual tree evaluations arise due to the"